oneview_redfish_toolkit/blueprints/computer_system.py

```diff
@@ -40,28 +41,27 @@ def get_computer_system(uuid):
     """Get the Redfish Computer System for a given UUID.
 
         Return ComputerSystem redfish JSON for a given
-        server hardware UUID.
-        Logs exception of any error and return abort(500)
-        Internal Server Error.
+        server hardware or server profile templates UUID.
+        Logs exception of OneViewRedfishError and abort(404).
 
         Returns:
             JSON: Redfish json with ComputerSystem
-            When Server hardware is not found calls abort(404)
-
-        Exceptions:
-            Logs the exception and call abort(500)
+            When Server hardware or Server profilte templates
+            is not found calls abort(404)
     """
-    try:
-        # Gets server hardware for given UUID
-        server_hardware = g.oneview_client.server_hardware.get(uuid)
 
-        # Gets the server hardware type of the given server hardware
-        server_hardware_types = g.oneview_client.server_hardware_types.get(
-            server_hardware['serverHardwareTypeUri']
-        )
-
-        # Build Computer System object and validates it
-        cs = ComputerSystem(server_hardware, server_hardware_types)
+    try:
+        resource = _get_oneview_resource(uuid)
+        category = resource["category"]
+        headers = ()
+
+        if category == 'server-hardware':
+            cs = _build_computer_system_server_hardware(resource)
+            headers = ("ETag", "W/" + resource['eTag'])
+        elif category == 'server-profile-templates':
+            cs = CapabilitiesObject(resource)
+        else:
+            raise OneViewRedfishError('Computer System type not found')
 
         # Build redfish json
         json_str = cs.serialize()
```

Comment:
We could use `ResponseBuilder.success()` in this case.

Reply:
Done.